agent/api/task/task.go

```diff
@@ -3723,3 +3723,23 @@ func (task *Task) HasActiveContainers() bool {
 	}
 	return false
 }
+
+func (task *Task) IsManagedDaemonTask() (string, bool) {
+	task.lock.RLock()
+	defer task.lock.RUnlock()
+
+	if !task.IsInternal {
+		return "", false
+	}
+
+	for _, c := range task.Containers {
+		containerStatus := c.GetKnownStatus()
+		if containerStatus.IsRunning() && c.IsInternal() {
```

Comment:
why does the container need to be running for us to check if it's name matches the "managed daemon" names?

Reply:
couldn't this be a race condition where the agent gets a new EBS task before the existing managed daemon task containers have gone to running?

Reply:
good question, we don't want to run into a scenario where the task/container was stopped but we still add it to the list of manged daemon tasks. otherwise agent won't be able to create/start a new CSI driver task prior to staging EBS volumes since it'll think that there's already an existing active CSI driver task

Reply:
> couldn't this be a race condition where the agent gets a new EBS task before the existing managed daemon task containers have gone to running?

we should be taking care of this when handling an EBS attachment payload -> https://github.com/aws/amazon-ecs-agent/blob/dev/agent/ebs/watcher.go#L132
we first check that if there's already an existing CSI driver task then we shouldn't create another. 

Thinking about it now, we should also include if there's a CSI driver task is being created

Reply:
One other thought about removing the IsRunning check: Managed Daemons have a container restart policy of `on failure` which implies that they may be running or stopped.  We should only ever have one container per daemon so we can track its health regardless of its process status.  We'll depend on the container restart policy to handle the daemon lifecycle and shouldn't launch redundant daemons.